Build failure: nifskope

### Nixpkgs version

- Unstable (25.05)

### Steps to reproduce

Build nifskope

### Can Hydra reproduce this build failure?

Yes, Hydra can reproduce this build failure.

### Link to Hydra build job

https://hydra.nixos.org/build/285772549

### Relevant log output

```console
lib/zlib/gzlib.c: In function 'gz_open':
lib/zlib/gzlib.c:14:17: error: implicit declaration of function 'lseek'; did you mean 'fseek'? [8;;https://gcc.gnu.org/onlinedocs/gcc-14.2.0/gcc/Warning-Options.html#index-Wimplicit-function-declaration-Wimplicit-function-declaration8;;]
   14 | #  define LSEEK lseek
      |                 ^~~~~
lib/zlib/gzlib.c:256:24: note: in expansion of macro 'LSEEK'
  256 |         state->start = LSEEK(state->fd, 0, SEEK_CUR);
      |                        ^~~~~
```

### Additional context

_No response_

### System metadata

 - system: `"x86_64-linux"`
 - host os: `Linux 6.12.9, NixOS, 25.05 (Warbler), 25.05.20250116.5df4362`
 - multi-user?: `yes`
 - sandbox: `yes`
 - version: `nix-env (Lix, like Nix) 2.91.1
System type: x86_64-linux
Additional system types: i686-linux, x86_64-v1-linux, x86_64-v2-linux, x86_64-v3-linux, x86_64-v4-linux
Features: gc, signed-caches
System configuration file: /etc/nix/nix.conf
User configuration files: /home/cassidy/.config/nix/nix.conf:/etc/xdg/nix/nix.conf:/home/cassidy/.nix-profile/etc/xdg/nix/nix.conf:/nix/profile/etc/xdg/nix/nix.conf:/home/cassidy/.local/state/nix/profile/etc/xdg/nix/nix.conf:/etc/profiles/per-user/cassidy/etc/xdg/nix/nix.conf:/nix/var/nix/profiles/default/etc/xdg/nix/nix.conf:/run/current-system/sw/etc/xdg/nix/nix.conf
Store directory: /nix/store
State directory: /nix/var/nix
Data directory: /nix/store/6dxshp2qaas92v7d36baf7y95q0v98dn-lix-2.91.1/share`
 - nixpkgs: `/nix/store/l9nb64iii15y0nr37qrs1cfm6rlpg6gh-source`

### Notify maintainers


---

**Note for maintainers:** Please tag this issue in your pull request description. (i.e. `Resolves #ISSUE`.)


### I assert that this issue is relevant for Nixpkgs

- [x] I assert that this is a bug and not a support request.
- [x] I assert that this is not a [duplicate of an existing issue](https://github.com/NixOS/nixpkgs/issues?q=is%3Aopen+is%3Aissue+label%3A%220.kind%3A+build+failure%22). 
- [x] I assert that I have read the [NixOS Code of Conduct](https://github.com/NixOS/.github/blob/master/CODE_OF_CONDUCT.md) and agree to abide by it.

### Is this issue important to you?

Add a :+1: [reaction] to [issues you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[issues you find important]: https://github.com/NixOS/nixpkgs/issues?q=is%3Aissue+is%3Aopen+sort%3Areactions-%2B1-desc

--- a/pkgs/tools/graphics/nifskope/default.nix
+++ b/pkgs/tools/graphics/nifskope/default.nix
@@ -7,6 +7,7 @@
   qttools,
   replaceVars,
   libGLU,
+  zlib,
   wrapQtAppsHook,
   fetchpatch,
 }:
@@ -19,12 +20,13 @@ stdenv.mkDerivation {
     owner = "niftools";
     repo = "nifskope";
     rev = "47b788d26ae0fa12e60e8e7a4f0fa945a510c7b2"; # `v${version}` doesn't work with submodules
-    sha256 = "1wqpn53rkq28ws3apqghkzyrib4wis91x171ns64g8kp4q6mfczi";
+    hash = "sha256-8TNXDSZ3okeMtuGEHpKOnKyY/Z/w4auG5kjgmUexF/M=";
     fetchSubmodules = true;
   };
 
   patches = [
     ./external-lib-paths.patch
+    ./zlib.patch
     (replaceVars ./qttools-bins.patch {
       qttools = "${qttools.dev}/bin";
     })
@@ -39,6 +41,7 @@ stdenv.mkDerivation {
     qtbase
     qttools
     libGLU
+    zlib
   ];
   nativeBuildInputs = [
     qmake

--- a/pkgs/tools/graphics/nifskope/zlib.patch
+++ b/pkgs/tools/graphics/nifskope/zlib.patch
@@ -1,0 +1,39 @@
+diff --git a/NifSkope.pro b/NifSkope.pro
+index 89b8471f6410..8e136ddf44c6 100644
+--- a/NifSkope.pro
++++ b/NifSkope.pro
+@@ -17,7 +17,7 @@ contains(QT_VERSION, ^5\\.[0-6]\\..*) {
+ CONFIG += c++14
+
+ # Dependencies
+-CONFIG += nvtristrip qhull zlib lz4 fsengine gli
++CONFIG += nvtristrip qhull lz4 fsengine gli
+
+ # Debug/Release options
+ CONFIG(debug, debug|release) {
+@@ -345,14 +345,6 @@ gli {
+     HEADERS += $$files($$PWD/lib/gli/external/glm/*.inl, true)
+ }
+
+-zlib {
+-    !*msvc*:QMAKE_CFLAGS += -isystem ./lib/zlib
+-    !*msvc*:QMAKE_CXXFLAGS += -isystem ./lib/zlib
+-    else:INCLUDEPATH += lib/zlib
+-    HEADERS += $$files($$PWD/lib/zlib/*.h, false)
+-    SOURCES += $$files($$PWD/lib/zlib/*.c, false)
+-}
+-
+ lz4 {
+     DEFINES += LZ4_STATIC XXH_PRIVATE_API
+
+@@ -442,6 +434,10 @@ win32 {
+     LIBS += -lopengl32 -lglu32
+ }
+
++unix {
++  LIBS += -lz
++}
++
+ unix:!macx {
+ 	LIBS += -lGLU
+ }

